app/Http/Middleware/Hotel/RedirectIfAuthenticated.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace HotelBooking\Http\Middleware\Hotel;
+
+use Closure;
+use Illuminate\Contracts\Auth\Guard;
+use Auth;
+use Route;
+
+/**
+ * Middleware class to redirect hotel admin out of login page if they have signed in
+ */
+class RedirectIfAuthenticated
+{
+    /**
+     * The Guard implementation.
+     *
+     * @var Guard
+     */
+    protected $auth;
+
+    /**
+     * Create a new filter instance.
+     *
+     * @param Guard $auth
+     */
+    public function __construct()
+    {
+        $this->auth = Auth::hotel();
+    }
+
+    /**
+     * Check if a hotel admin have logged in then redirect them to index page
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param \Closure                 $next
+     *
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        if ($this->auth->check()) {
+            if (Route::has('hotel.index')) {
+                return redirect(route('hotel.index'));
+            } else {
+                return 'This account already login';
```

Comment:
return 'This account already login' to make sure the page is redirect
successfull in case the route 'hotel.index' isn't exist.so the status code
should be 200, not 500(error page).

Vào Thứ Sáu, ngày 09 tháng 10 năm 2015, Le Mai Vien <
notifications@github.com> đã viết:
> In app/Http/Middleware/Hotel/RedirectIfAuthenticated.php:
>
>> +
>> + /**
>> + * Check if a hotel admin have logged in then redirect them to index
page
>> + *
>> + * @param \Illuminate\Http\Request $request
>> + * @param \Closure $next
>> + *
>> + * @return mixed
>> + */
>> + public function handle($request, Closure $next)
>> + {
>> + if ($this->auth->check()) {
>> + if (Route::has('hotel.index')) {
>> + return redirect(route('hotel.index'));
>> + } else {
>> + return 'This account already login';
>
> I don't understand this statement. I think we don't need it
>
> —
> Reply to this email directly or view it on GitHub.<
https://ci3.googleusercontent.com/proxy/PXw-m6A9qhPGMkbbcRjSHoGyhF8-vRx6g3-xMRAejqMipD8lP-JmxnPdRB4w7PR_dmiZEf9dug8-cd0UJB0l_OnHBlWklcT_ObjLjmBuKG5N_x6jjsJpMeWAYsLnP8xS02hBjG4RgucNFBtLpsboIaJFUvuBDQ=s0-d-e1-ft#https://github.com/notifications/beacon/AM_OMYoRP4FYJY3VcktzH_vyqqrupoL6ks5o52aPgaJpZM4GLzH4.gif
>